pkg/apis/kubeonecluster/types.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeonecluster
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeOneCluster is KubeOne Cluster API Schema
+type KubeOneCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec KubeOneClusterSpec `json:"spec,omitempty"`
+}
+
+// KubeOneClusterSpec is spec for the Cluster object
+type KubeOneClusterSpec struct {
+	Hosts        []*HostConfig `json:"hosts,omitempty"`
```

Comment:
I don't remember why originally it was `[]*HostConfig`, but probably we don't need that and `[]HostConfig` would be just fine.